modules/dnn/src/layers/recurrent_layers.cpp

```diff
@@ -359,9 +369,74 @@ class LSTMLayerImpl CV_FINAL : public LSTMLayer
                 Range curRowRange(ts*numSamples, (ts + 1)*numSamples);
                 Mat xCurr = xTs.rowRange(curRowRange);
 
-                gemm(xCurr, Wx, 1, gates, 0, gates, GEMM_2_T);      // Wx * x_t
-                gemm(hInternal, Wh, 1, gates, 1, gates, GEMM_2_T);  //+Wh * h_{t-1}
-                gemm(dummyOnes, bias, 1, gates, 1, gates);          //+b
+#if CV_TRY_AVX2
+                if (useAVX2 && xCurr.isContinuous() && gates.isContinuous() && bias.isContinuous() && Wx.depth() == CV_32F && xCurr.depth() == CV_32F && gates.depth() == CV_32F && bias.depth() == CV_32F && Wx.cols >= 8)
```

Comment:
could you please improve readability of the code as it does at the [fully connected layer](https://github.com/opencv/opencv/blob/master/modules/dnn/src/layers/fully_connected_layer.cpp#L231)? Condition 
```c++
gates.isContinuous() && bias.isContinuous() && gates.depth() == CV_32F && bias.depth() == CV_32F
```
could be calculated once. Or you can add a helper to ensure that Mats are continuous and have 32F type.